temperatures/routers.py

```diff
@@ -0,0 +1,29 @@
+from fastapi import APIRouter, Depends
+from sqlalchemy.ext.asyncio import AsyncSession
+from typing import List, Optional
+from dependencies import get_db
+from temperatures.schemas import TemperatureResponse
+from temperatures.crud import (
+    get_temperatures,
+    fetch_and_store_temperatures,
+    get_all_cities
+)
+
+
+router = APIRouter()
+
+
+@router.post("/temperatures/update/")
+async def update_temperatures(db: AsyncSession = Depends(get_db)):
+    cities = await get_all_cities(db)
+    await fetch_and_store_temperatures(db, cities)
+    return {"message": "Temperature data updated"}
+
+
+@router.get("/temperatures/", response_model=List[TemperatureResponse])
```

Comment:
The return type hint `List[TemperatureResponse]` requires importing `List` from the `typing` module, which is correctly done. Ensure that the Python version supports this syntax.